app/controllers/address/AddressListController.scala

```diff
@@ -61,19 +62,28 @@ trait AddressListController extends FrontendBaseController with I18nSupport with
       formWithErrors =>
         Future.successful(BadRequest(view(formWithErrors, viewModel, mode))),
       addressIndex => {
-        val address = viewModel.addresses(addressIndex).copy(country = Some("GB"))
-        val userAnswers = request.userAnswers.set(addressId)(address.toAddress).flatMap(
-          _.remove(postCodeLookupIdForCleanUp)).asOpt.getOrElse(request.userAnswers)
+        val address = viewModel.addresses(addressIndex).copy(countryOpt = Some("GB"))
+        if (address.toAddress.nonEmpty) {
```

Comment:
I think this could be simplified by means of fold or match to yield a tuple which then can be mapped. This would also get rid of the address.toAddress.get which isn't good practice.